lib/datadog/core/telemetry/component.rb

```diff
@@ -114,7 +119,7 @@ def integrations_change!
         end
 
         def log!(event)
-          return unless @enabled || forked?
+          return if !@enabled || forked? || !@log_collection_enabled
```

Comment:
I inverted this condition for the `forked?` boolean value, meaning it does not send logs on forked processes. This is the case for all telemetry signals: they don't send data on forks.

Reply:
I am pretty sure, telemetry logs are not sent on forks, due to the worker (the workers are not enabled after fork)

Reply:
I would put this note in the code, either here or in the setting description added earlier in the diff. Future readers of the code are unlikely to think to check PR comments for an explanation of why logging is not happening in a fork.

Reply:
I added a class-level comment to document that none telemetry is sent from forked processes.

Reply:
:+1: